### PR TITLE
[Geomagic] Add macro HAS_OPENHAPTICS to know if library is present

### PR DESCRIPTION
**Pull request by epernod**
Fix compilation without openHaptics to be able to activate it on the CI. 
Warning the GeomagicDriver will be invalid by default.

include PR #2113 

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
You should use instead the mechanism inside sofa_find_package() instead of adding a new macro definition.

You can see SofaGeneralEngine as an (easy) example.
CmakeLists.txt:
```cmake
sofa_find_package(Sofa.GL QUIET)
```
which will create the cmake variable SOFAGENERALENGINE_HAVE_SOFA_GL (${MODULENAME}_HAVE_${PACKAGENAME})

and in the config.h.in in the plugin, you can add:
```c++
#define SOFAGENERALENGINE_HAVE_SOFA_GL @SOFAGENERALENGINE_HAVE_SOFA_GL@
```
where @SOFAGENERALENGINE_HAVE_SOFA_GL@ will be either 0 or 1 (if the package were found in the cmake or not)
And then in your code, you can do:
```c++
#if SOFAGENERALENGINE_HAVE_SOFA_GL
// do stuff with Sofa.GL
#endif // SOFAGENERALENGINE_HAVE_SOFA_GL
```

In this case, you have a nicely define macro in the code and not defined with the complation commands (which add_definitions does with -**DHAVE_WHATEVER**)

File: applications/plugins/Geomagic/CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,9 @@ find_package(SofaLoader REQUIRED)
 find_package(OpenHaptics REQUIRED)
 
 if( NOT OPENHAPTICS_FOUND )
-    message( SEND_ERROR "OpenHaptics library not found, GeomagicDriver cannot compile")
+    message( WARNING "OpenHaptics library not found, GeomagicDriver cannot compile")
+elseif( NOT OPENHAPTICS_FOUND )
+    add_definitions("-DHAS_OPENHAPTICS")
```
Comment (fredroy) [inline applications/plugins/Geomagic/CMakeLists.txt]: See previous comment.

**Review by fredroy — CHANGES_REQUESTED** [review]

File: applications/plugins/Geomagic/CMakeLists.txt (hunk 1 above)
Comment (epernod) [reply applications/plugins/Geomagic/CMakeLists.txt]: so sofa_find_package((OpenHaptics QUIET) in Geomagic CMakeLists
will create GEOMAGIC_HAVE_OPENHAPTICS ?
how can I check the spelling  of: (${MODULENAME}HAVE${PACKAGENAME})

Comment (fredroy) [reply applications/plugins/Geomagic/CMakeLists.txt]: _"so sofa_find_package((OpenHaptics QUIET) in Geomagic CMakeLists
will create GEOMAGIC_HAVE_OPENHAPTICS ?"_
Yes

_"how can I check the spelling of: (${MODULENAME}HAVE${PACKAGENAME})"_
I guess in the SofaMacro in Sofa.Config but in your case it should be GEOMAGIC_HAVE_OPENHAPTICS

File: applications/plugins/Geomagic/src/Geomagic/config.h.in [hunk 2]
```diff
@@ -24,6 +24,8 @@
 
 #include <sofa/config.h>
 
+#define GEOMAGIC_HAVE_OPENHAPTICS @GEOMAGIC_HAVE_OPENHAPTICS@
+
```
Comment (guparan) [inline applications/plugins/Geomagic/src/Geomagic/config.h.in]:
```suggestion
#cmakedefine01 GEOMAGIC_HAVE_OPENHAPTICS

```

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by fredroy** [comment]
rebase (to clear the previous failing scenes) +
ignore scene from the Geomatic plugin

**Comment by epernod** [comment]
[ci-build]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by guparan** [comment]
I aborted the latest build because it must be triggered with a specific CI branch in which PLUGIN_GEOMAGIC is ON
Done in https://ci.inria.fr/sofa-ci-dev/job/sofa-framework/job/PR-2115/14/

**Review by guparan — APPROVED** [review]
CI looks happy, I approve :+1:

**Pull request merged**